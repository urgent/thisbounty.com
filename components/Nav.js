import React from "react";
import { signIn, signOut, useSession } from "next-auth/client";
import { Grid, Text, Box, Button } from "@chakra-ui/react";

export default function Nav() {
  const [session, loading] = useSession();

  return <View session={session} loading={loading} />;
}

export function View({ session, loading }) {
  return (
    <Grid
      as="nav"
      gridRow="nav"
      data-cy="nav"
      templateColumns="[home] 120px [user] 1fr"
      m={2}
    >
      {/* MenuItems are not rendered unless Menu is open */}
      <Box sx={{ gridColum: "home" }}>
        <a href="/">thisbounty.com</a>
      </Box>
      {!session && (
        <Box sx={{ gridColum: "user" }}>
          <Button
            data-cy="signIn"
            bg="brand.200"
            _hover={{ background: "brand.300" }}
            size="xs"
            onClick={() => signIn()}
          >
            Sign in
          </Button>
        </Box>
      )}
      {session && (
        <Box sx={{ gridColum: "user" }}>
          <Text fontSize="xs">Welcome, {session.user.name}</Text>
          <Button data-cy="signOut" onClick={() => signOut()}>
            Sign out
          </Button>
        </Box>
      )}
    </Grid>
  );
}
