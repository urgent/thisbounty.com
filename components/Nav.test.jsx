import React from "react";
import renderer from "react-test-renderer";
import { View } from "./Nav";

it("renders correctly", () => {
  const loggedOut = renderer.create(<View />).toJSON();
  expect(loggedOut).toMatchSnapshot();
  const loggedIn = renderer
    .create(<View session={{ user: { name: "test" } }} />)
    .toJSON();
  expect(loggedIn).toMatchSnapshot();
});
