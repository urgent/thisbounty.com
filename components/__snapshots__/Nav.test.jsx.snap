// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="css-i6ei9o"
  data-cy="nav"
>
  <div
    className="css-exb9x5"
  >
    <a
      href="/"
    >
      thisbounty.com
    </a>
  </div>
  <div
    className="css-ade8ji"
  >
    <button
      className="chakra-button css-1v83zyk"
      data-cy="signIn"
      onClick={[Function]}
    >
      Sign in
    </button>
  </div>
</nav>
`;
